lib/hooks.ts

```diff
@@ -30,13 +30,15 @@ export function useProducts() {
     };
 }
 
-export function useProductList() {
+export function useProductList(page?: number, limit?: number) {
     const encodedContext = useSession()?.context;
+    const query = `&page=${page}&limit=${limit}`;
     // Use an array to send multiple arguments to fetcher
-    const { data, error, mutate: mutateList } = useSWR(encodedContext ? ['/api/products/list', encodedContext] : null, fetcher);
+    const { data, error, mutate: mutateList } = useSWR(encodedContext ? [`/api/products/list`, encodedContext, query] : null, fetcher);
 
     return {
-        list: data,
+        list: data?.data,
```

Comment:
we don't need optional chaining for top-level props - please review documentation: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#syntax